retrofit/src/main/java/retrofit2/ParameterHandler.java

```diff
@@ -267,15 +304,23 @@
         builder.addFormField(entryKey, fieldEntry, encoded);
       }
     }
+
+    private void throwParameterError(String message) {
+      throw Utils.parameterError(method, p, message);
+    }
   }
 
   static final class Part<T> extends ParameterHandler<T> {
     private final Headers headers;
     private final Converter<T, RequestBody> converter;
+    private final Method method;
+    private final int p;
 
-    Part(Headers headers, Converter<T, RequestBody> converter) {
+    Part(Method method, int p, Headers headers, Converter<T, RequestBody> converter) {
       this.headers = headers;
       this.converter = converter;
+      this.method = method;
+      this.p = p;
```

Comment:
Can we make the field order and statement order match the constructor argument order?